### PR TITLE
Fixes the queue for beta 14

### PR DESCRIPTION
**Pull request by luceos**
A few changes were necessary to get the queue working again:

- New argument for maintenance flag in the Worker
- Overriding the WorkerCommand because it has a `Container` instance but assumes that is also a `Application` instance as it tries to call the `isDownForMaintenance` method. I'm resolving our Config.
- Removed the HackishQueueManager class.

---

Someone please test drive this locally as well:

`app/Extend/Queue.php`
```php
<?php

namespace App\Extend;

use Flarum\Extend\ExtenderInterface;
use Flarum\Extension\Extension;
use Illuminate\Contracts\Container\Container;
use Illuminate\Queue\NullQueue;

class Queue implements ExtenderInterface
{
    public function extend(Container $container, Extension $extension = null)
    {
        $container->extend('flarum.queue.connection', function () use ($container) {
            $queue = new NullQueue();
            $queue->setContainer($container);

            return $queue;
        });
    }
}
```
change in your `composer.json`
```json
    "autoload": {
        "psr-4": {
            "App\\": "app/"
        }
    },
```
`composer dumpautoload` to enable the new autoload directive.

This will re-enable the console commands for testing if you use this extender in the `extend.php` 👍 
```php

return [
    new App\Extend\Queue()
];
```

### CODE REVIEW

File: src/Queue/Console/WorkCommand.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class WorkCommand extends \Illuminate\Queue\Console\WorkCommand
+{
+    protected function downForMaintenance()
+    {
+        if ($this->option('force')) {
+            return false;
+        }
+
+        /** @var Config $config */
+        $config = $this->laravel->make(Config::class);
```
Comment (askvortsov1) [inline src/Queue/Console/WorkCommand.php:23]: Can we inject this?

File: src/Queue/QueueServiceProvider.php [hunk 2]
```diff
@@ -61,10 +62,16 @@ public function register()
         });
 
         $this->app->singleton(Worker::class, function ($app) {
+            /** @var Config $config */
+            $config = $app->make(Config::class);
+
             return new Worker(
-                new HackyManagerForWorker($app[Factory::class]),
```
Comment (askvortsov1) [inline src/Queue/QueueServiceProvider.php:65]: Is there a reason why we can use this now but couldn't before?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: src/Queue/Console/WorkCommand.php (hunk 1 above)
Comment (luceos) [reply src/Queue/Console/WorkCommand.php:23]: Yes but this is the least intrusive option available.

File: src/Queue/QueueServiceProvider.php (hunk 2 above)
Comment (luceos) [reply src/Queue/QueueServiceProvider.php:65]: Yes. We weren't able to override something. Not behind a computer but that's the bottom line.

**Review by tankerkiller125 — APPROVED** [review]

**Review by askvortsov1 — APPROVED** [review]
Alright, makes sense to me! Does this still need to be tested locally?

**Comment by datitisev** [comment]
Did anyone else test this locally? @luceos wanted someone else to test it before merging.

**Comment by askvortsov1** [comment]
yep!

**Pull request merged**